simd/unit_tests/include/TestSIMD_Reductions.hpp

```diff
@@ -37,7 +37,7 @@ inline void host_check_reduction_one_loader(ReductionOp reduce_op,
     if (!loaded_arg) continue;
 
     mask_type mask(false);
-    for (std::size_t j = 0; j < n; ++j) {
+    for (std::size_t j = 0; j < nlanes; ++j) {
```

Comment:
https://github.com/kokkos/kokkos/pull/7630 will avoid the issue with a generator expression.

Reply:
I think the same change should be applied on L114.

Reply:
FIxed.